apps/wallet/src/background/qredo/utils.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import Browser from 'webextension-polyfill';
+
+import { type QredoConnectInput } from '_src/dapp-interface/WalletStandardInterface';
+
+export function qredoConnectPageUrl(requestID: string) {
+    return `${Browser.runtime.getURL(
+        'ui.html'
+    )}#/dapp/qredo-connect/${encodeURIComponent(requestID)}`;
+}
+
+export function trimString(input: unknown) {
+    return (typeof input === 'string' && input.trim()) || '';
+}
+
+export function validateInputOrThrow(input: QredoConnectInput) {
+    if (!input) {
+        throw new Error('Invalid Qredo connect parameters');
+    }
+    let apiUrl;
+    try {
+        apiUrl = new URL(input.apiUrl);
+        if (!['http:', 'https:'].includes(apiUrl.protocol)) {
+            throw new Error('Only https (or http) is supported');
+        }
+    } catch (e) {
+        throw new Error(`Not valid apiUrl. ${(e as Error).message}`);
+    }
+    const service = trimString(input.service);
```

Comment:
do we need this `trimString` function since both `service` and `token` are strings and not `unknown`?

Reply:
It's done like this in case someone sends anything else from the dapp side (Probably there should be another layer of validation and we can skip this one but until then I just did it like this - [here](https://mysten.atlassian.net/browse/APPS-822) probably)